src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/tke_dissipation_equations.jl

```diff
@@ -0,0 +1,305 @@
+
+    # Patankar trick for TKE equation
+    wb⁻ = min(wb, zero(grid))
+    wb⁺ = max(wb, zero(grid))
+
+    eᵐⁱⁿ = closure_ij.minimum_tke
+    wb⁻_e = wb⁻ / eⁱʲᵏ * (eⁱʲᵏ > eᵐⁱⁿ)
+
+    # Patankar trick for ϵ-equation
+    Cᵋϵ = closure_ij.tke_dissipation_equations.Cᵋϵ
+    Cᵇϵ = closure_ij.tke_dissipation_equations.Cᵇϵ
+
+    Cᵇϵ_wb⁻ = min(Cᵇϵ * wb, zero(grid))
+    Cᵇϵ_wb⁺ = max(Cᵇϵ * wb, zero(grid))
+
+    # ∂t e = Lⁱ e + ⋯,
+    @inbounds Le[i, j, k] = wb⁻_e - ωe
+    @inbounds Lϵ[i, j, k] = Cᵇϵ_wb⁻ / e★ - Cᵋϵ * ωϵ
+
+    # Compute fast TKE and dissipation RHSs
+    u⁺ = next_velocities.u
+    v⁺ = next_velocities.v
+    uⁿ = previous_velocities.u
+    vⁿ = previous_velocities.v
+    κu = diffusivities.κu
+    Cᴾϵ = closure_ij.tke_dissipation_equations.Cᴾϵ
+
+    # TODO: correctly handle closure / diffusivity tuples
+    # TODO: the shear_production is actually a slow term so we _could_ precompute.
+    P = shear_production(i, j, k, grid, κu, uⁿ, u⁺, vⁿ, v⁺)
+
+    @inbounds begin
+        fast_Gⁿe = P + wb⁺                  # - ϵ (no implicit time stepping for now)
+        fast_Gⁿϵ = ωϵ * (Cᴾϵ * P + Cᵇϵ_wb⁺) 
+    end
+
+    # Advance TKE and store tendency
+    FT = eltype(χ)
+    Δτ = convert(FT, Δτ)
+
+    # See below.
+    α = convert(FT, 1.5) + χ
+    β = convert(FT, 0.5) + χ
+
+    @inbounds begin
+        total_Gⁿe = slow_Gⁿe[i, j, k] + fast_Gⁿe
+        total_Gⁿϵ = slow_Gⁿϵ[i, j, k] + fast_Gⁿϵ
+
+        e[i, j, k] += Δτ * (α * total_Gⁿe - β * G⁻e[i, j, k])
+        ϵ[i, j, k] += Δτ * (α * total_Gⁿϵ - β * G⁻ϵ[i, j, k])
+
+        G⁻e[i, j, k] = total_Gⁿe
+        G⁻ϵ[i, j, k] = total_Gⁿϵ
+    end
+end
+
+@inline function implicit_linear_coefficient(i, j, k, grid, closure::FlavorOfTD{<:VITD}, K, ::Val{id}, args...) where id
+    L = K._tupled_implicit_linear_coefficients[id]
+    return @inbounds L[i, j, k]
+end
+
+#####
+##### TKE top boundary condition
+#####
+
+@inline function top_tke_flux(i, j, grid, clock, fields, parameters, closure::FlavorOfTD, buoyancy)
+    closure = getclosure(i, j, closure)
+
+    top_tracer_bcs = parameters.top_tracer_boundary_conditions
+    top_velocity_bcs = parameters.top_velocity_boundary_conditions
+    tke_dissipation_parameters = closure.tke_dissipation_equations
+
+    return _top_tke_flux(i, j, grid, clock, fields, tke_dissipation_parameters, closure,
+                         buoyancy, top_tracer_bcs, top_velocity_bcs)
+end
+
+@inline function _top_tke_flux(i, j, grid, clock, fields,
+                               parameters::TKEDissipationEquations, closure::TDVD,
+                               buoyancy, top_tracer_bcs, top_velocity_bcs)
+
+    wΔ³ = top_convective_turbulent_velocity_cubed(i, j, grid, clock, fields, buoyancy, top_tracer_bcs)
+    u★ = friction_velocity(i, j, grid, clock, fields, top_velocity_bcs)
+
+    Cᵂu★ = parameters.Cᵂu★
+    CᵂwΔ = parameters.CᵂwΔ
+
+    return zero(grid) #- Cᵂu★ * u★^3 #- CᵂwΔ * wΔ³
```

Comment:
is this `zeros(grid)` + commented out code a left over from debugging or on purpose?

Reply:
Hmm yes, a non-zero TKE flux is non-canonical but we should support this with the parameters just chosen to be 0 for now

Reply:
OK, does this deserve a comment?